generators/csharp/sdk/src/options/IdempotentRequestOptionsGenerator.ts

```diff
@@ -0,0 +1,69 @@
+import { csharp, CSharpFile, FileGenerator } from "@fern-api/csharp-codegen";
+import { join, RelativeFilePath } from "@fern-api/fs-utils";
+import { SdkCustomConfigSchema } from "../SdkCustomConfig";
+import { SdkGeneratorContext } from "../SdkGeneratorContext";
+import { BaseOptionsGenerator } from "./BaseOptionsGenerator";
+
+export const IDEMPOTENT_REQUEST_OPTIONS_CLASS_NAME = "IdempotentRequestOptions";
+
+export class IdempotentRequestOptionsGenerator extends FileGenerator<
+    CSharpFile,
+    SdkCustomConfigSchema,
+    SdkGeneratorContext
+> {
+    private baseOptionsGenerator: BaseOptionsGenerator;
+
+    constructor(context: SdkGeneratorContext, baseOptionsGenerator: BaseOptionsGenerator) {
+        super(context);
+
+        this.baseOptionsGenerator = baseOptionsGenerator;
+    }
+
+    public doGenerate(): CSharpFile {
+        const class_ = csharp.class_({
+            ...this.context.getIdempotentRequestOptionsClassReference(),
+            partial: true,
+            access: csharp.Access.Public,
+            interfaceReferences: [this.context.getIdempotentRequestOptionsInterfaceClassReference()]
+        });
+        class_.addFields(this.baseOptionsGenerator.getRequestOptionFields());
+        class_.addFields(this.baseOptionsGenerator.getIdepotentRequestOptionFields());
+        class_.addMethod(
+            csharp.method({
+                name: "GetIdempotencyHeaders",
+                parameters: [],
+                return_: csharp.Type.reference(this.context.getHeadersClassReference()),
+                interfaceReference: this.context.getIdempotentRequestOptionsInterfaceClassReference(),
+                type: csharp.MethodType.INSTANCE,
+                body: csharp.codeblock((writer) => {
+                    writer.writeLine("return new Headers(new Dictionary<string, string>");
+                    writer.writeLine("{");
+                    writer.indent();
+                    for (const header of this.context.getIdempotencyHeaders()) {
+                        // TODO: find a way to check if a header is nullable, and add nullable check
```

Comment:
Is there an existing method to check whether something will be nullable?
- string
- object
- classes
- int?
- double?
but not
- int
- double
- etc.

I can't apply `?.` to everything as it won't compile for non-nullable types.

Reply:
can you check if the type of the header is a container + optional ?

Reply:
Don't see an existing method for it, but i could imagine it taking in a type reference and returning true/false